### PR TITLE
Share implementation of message handling on an `AsyncQueue` between `BuildSystemIntegration` and `SourceKitLSPServer`

### PR DESCRIPTION
**Pull request by ahoppen**
This gives us cancellation support for `BuiltInBuildSystemAdapter`.

### CODE REVIEW

**Comment by ahoppen** [comment]
@swift-ci Please test

File: Sources/SourceKitLSP/SourceKitLSPServer.swift [hunk 1]
```diff
@@ -42,22 +42,15 @@ package typealias Language = LanguageServerProtocol.Language
...
+  )
 
-  /// The queue on which we start and stop keeping track of cancellation.
-  ///
-  /// Having a queue for this ensures that we started keeping track of a
-  /// request's task before handling any cancellation request for it.
-  private let cancellationMessageHandlingQueue = AsyncQueue<Serial>()
+  package let messageHandlingQueue = AsyncQueue<MessageHandlingDependencyTracker>()
+
+  /// The queue on which
```
Comment (hamishknight) [inline Sources/SourceKitLSP/SourceKitLSPServer.swift:52]: Incomplete comment?

**Review by hamishknight — APPROVED** [review]

**Comment by ahoppen** [comment]
@swift-ci Please test Windows

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by ahoppen** [comment]
@swift-ci Please test Windows

**Pull request merged**